packages/presentation/src/panels/usePanelId.ts

```diff
@@ -1,7 +1,7 @@
-import {useRef} from 'react'
+import {useState} from 'react'
 import {v4 as uuid} from 'uuid'
 
 export function usePanelId(id?: string): string {
-  const idRef = useRef(id || uuid())
-  return idRef.current
+  const [panelId] = useState(() => id || uuid())
+  return panelId
```

Comment:
React Compiler doesn't like reading `.current` during render. In this case it happens to be safe.
A better alternative is to use `useState`. 
`useRef` itself is built on top of it:

```ts
export function useRef(initialValue) {
  const [ref] = useState({current: initialValue})
  return ref
}
```

In this case all we want is to ensure the `id` remains stable for the lifetime of the panel, we're never writing to the `current` property so there's no need to use `useRef` here, we can use `useState` instead.
A nice benefit here is that `useState` has a lazy init, so `uuid()` doesn't end up being called on every single render, it's only called on the first render if `id` is nullish.